--- ixgbe-4.1.2/src/ixgbe_main.c~	2015-09-27 11:47:47.000000000 +0200
+++ ixgbe-4.1.2/src/ixgbe_main.c	2015-09-27 21:07:30.561309223 +0200
@@ -2079,8 +2079,12 @@
 static inline bool ixgbe_page_is_reserved(struct page *page)
 {
 #ifdef HAVE_STRUCT_PAGE_PFMEMALLOC
+#if ( LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0) )
 	return (page_to_nid(page) != numa_mem_id()) || page->pfmemalloc;
 #else
+	return (page_to_nid(page) != numa_mem_id()) || page_is_pfmemalloc(page);
+#endif
+#else
 	return (page_to_nid(page) != numa_mem_id());
 #endif
 }
@@ -9473,7 +9473,11 @@
 	mode = adapter->bridge_mode;
 
 #ifdef HAVE_NDO_BRIDGE_GETLINK_NLFLAGS
+#if ( LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0) )
 	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0, nlflags);
+#else
+	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0, nlflags, filter_mask, NULL);
+#endif
 #elif defined(HAVE_NDO_FDB_ADD_VID)
 	return ndo_dflt_bridge_getlink(skb, pid, seq, dev, mode, 0, 0);
 #else
