--- ixgbe-3.21.2/src/kcompat.h~	2014-03-13 14:52:01.000000000 +0100
+++ ixgbe-3.21.2/src/kcompat.h	2014-07-12 11:34:56.647970376 +0200
@@ -4087,4 +4087,9 @@
 #define HAVE_NDO_SELECT_QUEUE_ACCEL_FALLBACK
 #endif /* 3.14.0 */
 
+#if ( LINUX_VERSION_CODE >= KERNEL_VERSION(3,15,0) )
+#define	u64_stats_fetch_begin_bh(a)	u64_stats_fetch_begin_irq(a)
+#define	u64_stats_fetch_retry_bh(a,b)	u64_stats_fetch_retry_irq(a,b)
+#endif
+
 #endif /* _KCOMPAT_H_ */
