--- ixgbe-4.1.5/src/kcompat.h~	2015-10-01 08:09:48.000000000 +0200
+++ ixgbe-4.1.5/src/kcompat.h	2015-10-01 08:11:39.877868999 +0200
@@ -4637,7 +4637,7 @@
 #define HAVE_NDO_BRIDGE_GETLINK_NLFLAGS
 #endif /* 4,1,0 */
 
-#if ( LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0) )
+#if ( (LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0)) && !(LINUX_VERSION_CODE >= KERNEL_VERSION(4,1,9)) )
 #if (!(SLE_VERSION_CODE && SLE_VERSION_CODE >= SLE_VERSION(12,1,0)))
 static inline bool page_is_pfmemalloc(struct page __maybe_unused *page)
 {
@@ -4650,8 +4650,10 @@
 #endif /* !SLES12sp1 */
 
 #else
-#define HAVE_NDO_DFLT_BRIDGE_GETLINK_VLAN_SUPPORT
 #undef HAVE_STRUCT_PAGE_PFMEMALLOC
 #endif /* 4.2.0 */
+#if ( LINUX_VERSION_CODE >= KERNEL_VERSION(4,2,0) )
+#define HAVE_NDO_DFLT_BRIDGE_GETLINK_VLAN_SUPPORT
+#endif /* 4.2.0 */
 
 #endif /* _KCOMPAT_H_ */
